stdlib/SparseArrays/src/sparsevector.jl

```diff
@@ -378,13 +378,10 @@ sparsevec(a::AbstractSparseArray) = vec(a)
 sparsevec(a::AbstractSparseVector) = vec(a)
 sparse(a::AbstractVector) = sparsevec(a)
 
-function _dense2sparsevec(s::AbstractArray{Tv}, initcap::Ti) where {Tv,Ti}
+function _dense2indval!(nzind::Vector{Ti}, nzval::Vector{Tv}, s::AbstractArray{Tv}) where {Tv,Ti}
     @assert !has_offset_axes(s)
-    # pre-condition: initcap > 0; the initcap determines the index type
+    cap = length(nzind);
```

Comment:
Let's also assert that `length(nzind) == length(nzval)` here as we assume that and use `@inbounds` with that assumption. It's an internal function so we needn't be too kind with the error message.

```suggestion
    cap = length(nzind);
    @assert cap == length(nzval)
```